app/scripts/controllers/main.js

```diff
@@ -0,0 +1,224 @@
+  {
+    $scope.trial = "hello"
+    console.log("controller");
+      //$scope.number = 17;
+      $scope.type = 'user';
+      $scope.comments = 'Number of Tracks';
+      $scope.colour = "primary";
+      var url1 = "https://envirocar.org/api/stable/users/";
+      url1 = url1 + $rootScope.globals.currentUser.username + "/tracks";
+      $http.defaults.headers.common = {'X-User': $rootScope.globals.currentUser.username, 'X-Token': $rootScope.globals.currentUser.authdata};
+      console.log("url is"+url1 )
+      var timeline = {};
+
+      requesthomestats.get(url1).then(function(data){
+        $scope.number = data.data.tracks.length;
+        var limit=0;
+        if($scope.number>=5)
+          limit=5;
+        else
+          limit = $scope.number;
+        for(var i=0; i<limit; i++)
+        {
+          (function(cntr){
+                requesthomestats.get(url1+"/"+data.data.tracks[cntr].id).then(function(data2){
+                  console.log(data2.data);
+                  data2.data['url']="https://envirocar.org/api/stable/tracks/"+data2.data.properties.id+"/preview";
+                  data2.data['urlredirect'] = "#/dashboard/chart?trackid="+data2.data.properties.id;
+                  //data2.data['urlredirect'] = "chart?trackid="+data2.data.properties.id;
+
+                  timeline[data.data.tracks[cntr].id] = data2.data;
+                //  console.log(timeline[data.data.tracks[cntr].id]);
+                })
+          })(i);
+          //console.log(timeline);
+
+        }
+          //now to get the data required for the repeater
+
+      });
+      console.log(timeline);
+      $scope.timelinevalues = timeline;
+
+
+      //*******************************************************
+      //*******************************************************
+      //******************GRAPHS PART**************************
+      $scope.options = {
+              chart: {
+                  type: 'multiBarHorizontalChart',
+                  height: 450,
+                  x: function(d){return d.label;},
+                  y: function(d){return d.value;},
+                  //yErr: function(d){ return [-Math.abs(d.value * Math.random() * 0.3), Math.abs(d.value * Math.random() * 0.3)] },
+                  showControls: false,
+                  showValues: true,
+                  duration: 500,
+                  xAxis: {
+                      showMaxMin: false
+                  },
+                  yAxis: {
+                      axisLabel: 'Values',
+                      tickFormat: function(d){
+                          return d3.format(',.2f')(d);
+                      }
+                  }
+              }
+          };
+          var datausers = [];
+          var dataotherusers =[];
+            var url = "https://envirocar.org/api/stable/users/"+$rootScope.globals.currentUser.username +"/statistics/CO2";
```

Comment:
I recommend to encapsulate constant values like the server API url to a standalone angular module. The standalone module acts as a configuration module and can be injected in other modules, thus making it available globally. This way we do not need to define stuff multiple times and it is also very clean.

Some reading regarding constant modules: http://stackoverflow.com/questions/28416054/inject-constant-to-other-modules-config-using-angular-js#answer-28416656